hcl2template/types.packer_config.go

```diff
@@ -362,7 +362,11 @@ func (cfg *PackerConfig) evaluateDatasources(skipExecution bool) hcl.Diagnostics
 			continue
 		}
 
+		dsOpts, _ := decodeHCL2Spec(body, cfg.EvalContext(DatasourceContext, nil), datasource)
+
```

Comment:
nit: We can drop the empty line here, as dsOpt is related to the next statement. 
```suggestion
```